src/ReactTags.js

```diff
@@ -272,9 +273,14 @@ class ReactTags extends Component {
     }
 
     e.preventDefault();
+    // Checking if the pasted text is greater than maxLength props
+    const {maxLength} = this.props;
 
     const clipboardData = e.clipboardData || window.clipboardData;
-    const pastedText = clipboardData.getData('text');
+    const clipboardText = clipboardData.getData('text');
+
+    const maxTextLength = Math.min(maxLength, clipboardText.length);
```

Comment:
instead of setting `maxLength` as Infinity can we do one of the below
*  `const { maxLength = clipboardText.length } = this.props;`
*  `const maxTextLength = Math.min(maxLength, clipboardText.length); || clipboardText.length`